src/services/githubauth.service.js

```diff
@@ -124,6 +125,44 @@ export default class GithubAuthenticationService {
     return tokenCreated;
   }
 
+  /**
+   * First exchanges code and state for a github token.
+   * Then adds a github token for the given user into the githubTokens table. If a token already
+   * exists for the user, it will automatically update the old one.
+   * @param userId the user id to add the token for
+   * @param code the code to exchange for token
+   * @param state the state to verify identity
+   * @param username the github username of the user
+   * @returns {Object} the token that was created
+   */
+  async getAndSetGithubTokenForUser(userId, code, state, username) {
+    try {
+      const secrets = config.get('oauth_secrets');
+      const res = await fetch('https://cors-anywhere.herokuapp.com/https://github.com/login/oauth/access_token', {
+        method: 'POST',
+        body: JSON.stringify({
+          client_id: '6b57706945fa18ee0397',
+          client_secret: secrets.github,
+          code,
+          state
+        }),
+        headers: new fetch.Headers({
+          'Content-Type': 'application/json',
+          origin: 'http://localhost:8080/',
+          Accept: 'application/json'
+        }),
+      });
+      if (res.status < 400) {
+        const data = await res.text().then(text => (text ? JSON.parse(text) : null));
```

Comment:
Can you add a comment to what this code is doing?

Why is it necessary to get the text, then parse the JSON from the text?

Reply:
it's how the front end does this so that it doesn't implode when the json is empty for whatever reason

Reply:
ugh. That's annoying. Can you write a comment about it :)

Reply:
We can leave it.

Reply:
If it's used in three places, perhaps it should be moved to a util file function. It would be much cleaner to read something like `const data = parseJsonPromise(res.text());`